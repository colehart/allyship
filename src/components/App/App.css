.App {
  height: 100vh;
  width: 100vw;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 20% 1fr;
}

.a-404 {
  width: 75%;
  margin: 0 auto;
  grid-row: 2 / 3;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 40px 0;
  max-width: 550px;
}

.a-404 h2 {
  text-align: center;
}

.a-404 p {
  font-size: 1.15rem;
}