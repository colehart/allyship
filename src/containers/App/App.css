.App {
  display: grid;
  grid-template-columns: 1vw;
  grid-template-rows: 30% 1fr;
}
