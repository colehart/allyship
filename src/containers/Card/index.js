import React from 'react';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { toggleSaved } from '../../actions';
import './Card.css';

export const Card = (props) => {
  const {
    source,
    author,
    title,
    description,
    url,
    urlToImage,
    published,
    isSaved,
    toggleSaved
  } = props

  return (
    <article className='Card'>
      <div className='c-title-group'>
        <h3 className='c-title'>{title}</h3>
        <div
          className={`c-save ${ isSaved ? 'is-saved' : '' }`}
          onClick={() => toggleSaved(title)}
        >
        </div>
      </div>
      <p className='c-byline'>{published} | {source} {`${author ? `| ${author}` : ''}`}</p>
      <img className='c-image' src={urlToImage} alt='provided illustration'/>
      <p className='c-description'>{description}</p>
      <a
        className='c-link'
        href={url}
        target='_blank'
        rel='noopener noreferrer'
      >
        Read article from source
      </a>
    </article>
  )
}

export const mapDispatchToProps = (dispatch) => ({
  toggleSaved: (headline) => dispatch(toggleSaved(headline))
})

Card.propTypes = {
  toggleSaved: PropTypes.func
}

export default connect(null, mapDispatchToProps)(Card)