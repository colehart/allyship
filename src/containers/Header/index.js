import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import shuttleIcon from '../../assets/images/space-shuttle.svg';
import './Header.css';

export const Header = () => {
  return (
    <div className='Header'>
      <Link
        to='/'
        className='h-logo-group'
      >
        <img
          className='h-logo'
          src={ shuttleIcon }
          alt='A space shuttle'
        />
        <h1>Allyship</h1>
      </Link>
      <nav className='h-nav-group'>
        <NavLink
          exact to='/'
          className="h-categories btn"
        >
          Categories
        </NavLink>
        <NavLink
          exact to='/saved'
          className="h-saved btn"
        >
          Saved
          <span className='h-badge'> 0</span>
        </NavLink>
      </nav>
    </div>
  )
}

export const mapStateToProps = (state) => ({
  caughtError: state.caughtError,
  stories: state.stories
})

export default connect(mapStateToProps)(Header)