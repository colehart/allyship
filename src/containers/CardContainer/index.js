import React from 'react';
import Card from '../Card'
import './CardContainer.css';

export const CardContainer = () => {
  const cards = []

  return (
    <main className="CardContainer">
      { cards }
    </main>
  )
}
