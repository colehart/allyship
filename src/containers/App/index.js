import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { Header } from '../Header';
import { Welcome } from '../Welcome';
import { CardContainer } from '../CardContainer';
import './App.css';

export const App = (props) => {
  const { isLoading } = props

  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Your ship has gone off course.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the orange buttons in this app to navigate to your desired destination.</p>
    </div>
  )

  return (
    <div className='App'>
      <Header />
      <Switch>
        <Route
          exact path='/'
          component={ Welcome }
        />
        <Route
          exact path='/saved'
          render={() => CardContainer }
        />
        <Route
          exact path='/transgender'
          render={() => CardContainer }
        />
        <Route
          exact path='/immigration'
          render={() => CardContainer }
        />
        <Route
          exact path='/black-lives-matter'
          render={() => CardContainer }
        />
        <Route
          component={noMatch}
        />
      </Switch>
      {/* <Route
        exact path='/search'
        render={() => CardContainer }
      /> */}
    </div>
  );
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading,
})

export default connect(mapStateToProps)(App);
