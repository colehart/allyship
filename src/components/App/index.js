import React from 'react';
import { Route, Switch } from 'react-router-dom';
import { Header } from '../../containers/Header';
import { Welcome } from '../../containers/Welcome';
import { CardContainer } from '../../containers/CardContainer';
import './App.css';

export const App = () => {
  const noMatch = ({ location }) => (
    <div className='a-404'>
      <h2>404 - Your ship has gone off course.</h2>
      <p>No routes match {location.pathname}.</p>
      <p>Please click the orange buttons in this app to navigate to your desired destination.</p>
    </div>
  )

  return (
    <div className='App'>
      <Header stories={[]}/>
      <Switch>
        <Route
          exact path='/'
          component={ Welcome }
        />
        <Route
          exact path='/saved'
          render={ () => <CardContainer /> }
        />
        <Route
          exact path='/transgender'
          render={ () => <CardContainer /> }
        />
        <Route
          exact path='/immigration'
          render={ () => <CardContainer /> }
        />
        <Route
          exact path='/black-lives-matter'
          render={ () => <CardContainer /> }
        />
        <Route
          component={noMatch}
        />
      </Switch>
      {/* <Route
        exact path='/search'
        render={() => CardContainer }
      /> */}
    </div>
  );
}