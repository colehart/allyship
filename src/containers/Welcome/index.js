import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { PropTypes } from 'prop-types';
import { Loader } from '../../components/Loader';
import { fetchStories } from '../../thunks/fetchStories';
import './Welcome.css';

export const Welcome = (props) => {
  const { isLoading } = props

  return (
    <main className='Welcome'>
      { isLoading ? <Loader /> : null }
      <section
        className={ isLoading ? 'hidden' : 'w-message'}
      >
        <h2>Welcome!</h2>
        <p>
          Allyship exists to help you find your way in the galaxy of knowledge and news any true ally must navigate.
        <br />
        <br />
          Research the latest news about a specific minority identity or issue.
        {/* , or choose your own query by entering a search keyword or phrase. */}
        </p>
      </section>
      <div
        className='w-category-group'
      >
        <Link
          to='/transgender'
          exact='true'
          className='btn'
          aria-label='Click here for the latest transgender news'
        >
          Transgender
        </Link>
        <Link
          to='/immigration'
          exact='true'
          className='btn'
          aria-label='Click here for the latest immigration news'
        >
          Immigration
        </Link>
        <Link
          to='/black-lives-matter'
          exact='true'
          aria-label='Click here for the latest black lives matter news'
          className='btn'
        >
          Black Lives Matter
        </Link>
      </div>
    </main>
  )
}

export const mapStateToProps = (state) => ({
  isLoading: state.isLoading
})

export const mapDispatchToProps = (dispatch) => ({
  fetchStories: (url) => dispatch(fetchStories(url))
})

Welcome.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  fetchStories: PropTypes.func.isRequired,
}

export default connect(mapStateToProps, mapDispatchToProps)(Welcome)
