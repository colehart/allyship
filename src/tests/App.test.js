import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { App, mapStateToProps } from '../containers/App';
import { mockDefaultState } from './testMocks'
describe('App', () => {
  describe('App Component', () => {
    let wrapper;

    beforeEach(() => {
      wrapper = shallow(<App/>)
    })

    it('matches the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })

  describe('matchStateToProps', () => {
    it('should parse isLoading from state', () => {
      const expected = false

      const mappedProps = mapStateToProps(mockDefaultState)
      expect(mappedProps.isLoading).toEqual(expected)
    })
  })
})