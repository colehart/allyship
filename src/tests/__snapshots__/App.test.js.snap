// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App App Component matches the snapshot 1`] = `
<div
  className="App"
>
  <Header />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      exact={true}
      path="/saved"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/transgender"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/immigration"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/black-lives-matter"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</div>
`;
